src/cpp/r/session/graphics/RShadowPngGraphicsHandler.cpp

```diff
@@ -192,6 +241,23 @@ Error shadowDevDesc(DeviceContext* pDC, pDevDesc* pDev)
          Error err = r::exec::executeString(code);
          if (err)
             return err;
+         
+         // stop-gap fix for crash with usages of newer graphics device APIs
+         // added in R 4.1.0 / graphics engine 14
+         // https://github.com/rstudio/rstudio/issues/9251
+         if (R_GE_getVersion() >= 14)
```

Comment:
This is really the "core" of the PR -- we override these callbacks on the R graphics device. See https://github.com/rstudio/rstudio/issues/9251#issuecomment-825412073 for extra motivation.